File: Algorithms/sorting/insertion_sort/insertion_sort.js
const insertion_sort = (array) => {
	for (let i = 0; i < array.length; i++) {
		const key = array[i];

		let j = i - 1;

		while (j >= 0 && array[j] > key) {
			array[j + 1] = array[j];

			j--;
		}

		array[j + 1] = key;
	}
};

a = [1, 3, 2, 0, 5, 4, -1];

console.log(a);

insertion_sort(a);

console.log(a);

File: Algorithms/sorting/insertion_sort/insertion_sort.java
package Algorithms.sorting.insertion_sort;

public class insertion_sort {

    static void printList(int[] arr) {
        for (int i : arr) {
            System.out.print(i + " ");
        }
        System.out.println("\n");
    }

    static void insertionSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int key = arr[i];

            int j = i - 1;

            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];

                j--;
            }

            arr[j + 1] = key;
        }
    }

    public static void main(String[] args) {
        int[] arr = { 1, 3, 2, 0, 5, 4, -1 };

        printList(arr);

        insertionSort(arr);

        printList(arr);

    }
}
